lib/data/file/backend.js

```diff
@@ -37,40 +37,38 @@ export const backend = {
         // Consider making async
         const key = crypto.randomBytes(20).toString('hex');
         const filePath = getFilePath(key);
-
         request.pause();
-
-        fs.open(filePath, 'wx', (err, fd) => {
+        fs.open(filePath, 'w', (err, fd) => {
             if (err) {
                 log.error('error opening filePath', { error: err });
                 return callback(errors.InternalError);
             }
+            const fileStream = fs.createWriteStream(filePath, { fd });
+
             request.resume();
-            return request.on('data', data => {
-                // Disable data events as we need to wait for fs.write callback
-                request.pause();
-                return fs.write(fd, data, 0, data.length,
-                    err => {
-                        if (err) {
-                            log.error('error writing data', { error: err });
-                            fs.close(fd);
-                            return callback(errors.InternalError);
-                        }
-                        return request.resume(); // Allow data events again
-                    });
-            })
-            .on('error', err => {
-                log.error('error streaming data from request', { error: err });
-                fs.close(fd);
+            request.pipe(fileStream, { end: false }).on('error', err => {
+                log.error('error streaming data from request on write',
+                    { error: err });
                 return callback(errors.InternalError);
-            })
-            .on('end', () => {
-                log.debug('finished writing data', { key });
-                fs.fsync(fd, () => {
-                    fs.close(fd);
-                    return callback(null, key);
+            });
+            request.on('error', err => {
+                log.error('error streaming data from request on read',
+                    { error: err });
+                // close fileStream
+                return fs.close(fd, () => callback(errors.InternalError));
+            }).on('end', () => {
```

Comment:
The next part should be after the ```finish``` event of the writable stream . 
And i don't understand why not ending the writableStream at the end ?
```js
request.pipe(fileStream).on('error', err => {
    // log your error and all 
}).on('finish', () => {
       fs.fsync(fd, err => {
             if (err) {
                 log.error('error streaming data from request on fsync',
                     { error: err });
                 return callback(errors.InternalError);
             }
             log.debug('finished writing data', { key });
             return fs.close(fd, () => callback(null, key));
        });
});
```